src/js/video.js

```diff
@@ -96,10 +96,70 @@ function videojs(id, options, ready) {
   }
 
   // Element may have a player attr referring to an already created player instance.
-  // If not, set up a new player and return the instance.
-  return tag.player || Player.players[tag.playerId] || new Player(tag, options, ready);
+  // If so return that otherwise set up a new player below
+  if (tag.player || Player.players[tag.playerId]) {
+    return tag.player || Player.players[tag.playerId];
+  }
+
+  videojs.hooks('presetup').forEach(function(hookFunction) {
```

Comment:
The `ModalDialog` component has similar lifecycle events, but the naming scheme is:

- `beforefoo`: Happens before the thing.
- `foo`: Happens after the thing.

Maybe we ought to follow that universally and use `beforesetup` and `setup`?